File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    answer = [] 
    brown_height = []
    brown_width = []
    size = brown + yellow
    end = int(size**0.5)
    # 약수 짝 다 찾고
    for i in range(1, end+1):
        if size % i == 0:
            brown_height.append(i)
            brown_width.append(size/i)

    # 거기서 -2 씩 뺀게 yellow랑 같은지 확인
    for h, w in zip(brown_height, brown_width):
        if (h-2) * (w-2) == yellow:
            return [w, h]
    return answer

File: 프로그래머스/2/42842. 카펫/카펫.swift
import Foundation
/**
격자의 수 == 넓이
직사각형의 넓이의 약수 짝 -> 가로, 세로
(y_w + y_h) * 2 + 4(= 모서리 네 개) == brown의 넓이
*/
func solution(_ brown:Int, _ yellow:Int) -> [Int] {
    let pairOfDivisors = getPairOfDivisors(yellow)
    var result = [Int]()
    
    for (w, h) in pairOfDivisors {
        if (w + h) * 2 + 4 == brown {
            result = [w + 2, h + 2]
        }
    }
    
    return result
}

func getPairOfDivisors(_ num: Int) -> [(w: Int, h: Int)] {
    let end = Int(sqrt(Double(num)))
    var result = [(Int, Int)]()
    
    for i in 1...end {
        if num % i == 0 {
            result.append((num / i, i))
        }
    }
    
    return result
}